mllib/src/main/scala/org/apache/spark/ml/optim/WeightedLeastSquares.scala

```diff
@@ -137,61 +133,68 @@ private[ml] class WeightedLeastSquares(
       }
     }
 
-    // scale aBar to standardized space in-place
-    val aBarValues = aBar.values
-    var j = 0
-    while (j < numFeatures) {
-      if (aStd(j) == 0.0) {
-        aBarValues(j) = 0.0
-      } else {
-        aBarValues(j) /= aStd(j)
-      }
-      j += 1
-    }
+    val bBar = summary.bBar / bStd
+    val bbBar = summary.bbBar / (bStd * bStd)
 
-    // scale abBar to standardized space in-place
-    val abBarValues = abBar.values
-    val aStdValues = aStd.values
-    j = 0
-    while (j < numFeatures) {
-      if (aStdValues(j) == 0.0) {
-        abBarValues(j) = 0.0
-      } else {
-        abBarValues(j) /= (aStdValues(j) * bStd)
+    val aStd = summary.aStd
+    val aBar = {
+      val _aBar = summary.aBar
+      var i = 0
+      // scale aBar to standardized space in-place
+      while (i < numFeatures) {
+        if (aStd(i) == 0.0) {
+          _aBar.values(i) = 0.0
```

Comment:
You're probably right. The only thing it might have a real impact is `aaBar`, but since we use while loops for speed, it seems better to also use this optimization as well.